sk-usbhid.c

```diff
@@ -487,6 +504,13 @@ sk_select_by_cred(const fido_dev_info_t *devlist, size_t ndevs,
 		skv[0] = NULL;
 		goto out;
 	}
+	for (i = 0; i < skvcnt; i++) {
+		if (fido_dev_is_winhello (skv[i]->dev)) {
+			sk = skv[i];
+			skv[i] = NULL;
+			goto out;
+		}
+	}
```

Comment:
I see your point. When I created this patch, the idea was to prefer WinHello for the admin user as well. The reason is this: On Windows since Vista, the admin user is usually not a special account anymore. We have users with administrative privileges, which are by default not present in the user token. If the user wants to perform admin tasks, the user starts a process with admin privileges explicitely, typically in the same session.

What I'm up to here is, without my change, calling ssh in an admin session and calling ssh in a non-admin session, even on the same desktop, would use different UIs.  The admin session would prefer direct device access, the non-admin session is restricted to WinHello.

From a UI perspective, this is a bad interface.

My question here is, how important is ssh-keygen -K? In theory, the user wants to use the token, that's what FIDO2 is about. I'm aware of the potential deficits, but I think a consistent UI is preferrable over easy usage of seldomly used  functionality.

Reply:
I don't think Pedro's proposal yields inconsistent UI. For the default case (no `-Odevice="..."` specified), it will use the WinHello device. A physical token could only be selected if it's an admin **and** they explicitly request a device